lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/LibertyProjectsManager.java

```diff
@@ -63,7 +64,13 @@ public String getLibertyVersion(LibertyWorkspace libertyWorkspace) {
      */
     public LibertyWorkspace getWorkspaceFolder(String serverXMLUri) {
         for (LibertyWorkspace folder : getInstance().getLibertyWorkspaceFolders()) {
-            if (serverXMLUri.contains(folder.getWorkspaceString())) {
+            //Append workspaceDirUri with file separator to avoid bad matches
+            String workspaceDirUri = folder.getWorkspaceString();
+            if (!workspaceDirUri.endsWith(File.separator)) {
```

Comment:
Can `workspaceDirUri` be null? Just wondering because of the NPE in the Windows build.

Reply:
Yeah, some of the tests don't build the full workspace since the initializeParams are null.

Reply:
Not sure if things would get this far since the workspace folders should be empty in that case.

Reply:
Nevermind, looks like this test does set some workspaces, so it might be null there then.